src/containerapp/azext_containerapp/_utils.py

```diff
@@ -2,21 +2,218 @@
 # Copyright (c) Microsoft Corporation. All rights reserved.
 # Licensed under the MIT License. See License.txt in the project root for license information.
 # --------------------------------------------------------------------------------------------
-# pylint: disable=line-too-long, consider-using-f-string, no-else-return, duplicate-string-formatting-argument
+# pylint: disable=line-too-long, consider-using-f-string, no-else-return, duplicate-string-formatting-argument, expression-not-assigned, too-many-locals, logging-fstring-interpolation
+
+import time
+import json
+import platform
 
 from urllib.parse import urlparse
-from azure.cli.command_modules.appservice.custom import (_get_acr_cred)
-from azure.cli.core.azclierror import (ValidationError, RequiredArgumentMissingError)
+from datetime import datetime
+from dateutil.relativedelta import relativedelta
+from azure.cli.core.azclierror import (ValidationError, RequiredArgumentMissingError, CLIInternalError,
+                                       ResourceNotFoundError, ArgumentUsageError)
 from azure.cli.core.commands.client_factory import get_subscription_id
 from knack.log import get_logger
-from msrestazure.tools import parse_resource_id
+from msrestazure.tools import parse_resource_id, is_valid_resource_id, resource_id
 
 from ._clients import ContainerAppClient
 from ._client_factory import handle_raw_exception, providers_client_factory, cf_resource_groups, log_analytics_client_factory, log_analytics_shared_key_client_factory
 
 logger = get_logger(__name__)
 
 
+# original implementation at azure.cli.command_modules.role.custom.create_service_principal_for_rbac
+# reimplemented to remove unnecessary warning statements
+def create_service_principal_for_rbac(  # pylint:disable=too-many-statements,too-many-locals, too-many-branches, unused-argument, inconsistent-return-statements
+        cmd, name=None, years=None, create_cert=False, cert=None, scopes=None, role=None,
+        show_auth_for_sdk=None, skip_assignment=False, keyvault=None):
+    from azure.cli.command_modules.role.custom import (_graph_client_factory, TZ_UTC, _process_service_principal_creds,
+                                                       _validate_app_dates, create_application,
+                                                       _create_service_principal, _create_role_assignment,
+                                                       _error_caused_by_role_assignment_exists)
```

Comment:
Importing and using private method is definitely a no-go!

Reply:
@jiasli this is something we need for the initial release of the extension for 'up' support &  we have to copy this over since we want some additional logger.warning we want to add. We will move to the new graph calls with the next release, Could you point us to the new Microsoft.Graph migration changes as well, so we can change it to that & do testing for 'up' command? Thanks!

Reply:
I found this https://docs.microsoft.com/en-us/cli/azure/microsoft-graph-migration?tabs=powershell#az-ad-app-create @StrawnSC I will add this to our ADO work item so we understand the exact changes we need to make next to ensure we use Microsft.Graph implementation to add SP.

Reply:
This extension will break with `azure-cli-core` 2.37.0. Please see the upper bound of `azure-cli-core` to 2.36.0.